packages/analytics/lib/modular/index.d.ts

```diff
@@ -0,0 +1,1188 @@
+import Analytics = FirebaseAnalyticsTypes.Module;
+import AnalyticsCallOptions = FirebaseAnalyticsTypes.AnalyticsCallOptions;
+import AnalyticsSettings = FirebaseAnalyticsTypes.AnalyticsSettings;
+import BeginCheckoutEventParameters = FirebaseAnalyticsTypes.BeginCheckoutEventParameters;
+import CampaignDetailsEventParameters = FirebaseAnalyticsTypes.CampaignDetailsEventParameters;
+import CustomEventName = FirebaseAnalyticsTypes.CustomEventName;
+import EarnVirtualCurrencyEventParameters = FirebaseAnalyticsTypes.EarnVirtualCurrencyEventParameters;
+import EventParams = FirebaseAnalyticsTypes.EventParams;
+import FirebaseApp = ReactNativeFirebase.FirebaseApp;
+import GenerateLeadEventParameters = FirebaseAnalyticsTypes.GenerateLeadEventParameters;
+import JoinGroupEventParameters = FirebaseAnalyticsTypes.JoinGroupEventParameters;
+import LevelEndEventParameters = FirebaseAnalyticsTypes.LevelEndEventParameters;
+import LevelStartEventParameters = FirebaseAnalyticsTypes.LevelStartEventParameters;
+import LevelUpEventParameters = FirebaseAnalyticsTypes.LevelUpEventParameters;
+import LoginEventParameters = FirebaseAnalyticsTypes.LoginEventParameters;
+import PostScoreEventParameters = FirebaseAnalyticsTypes.PostScoreEventParameters;
+import PurchaseEventParameters = FirebaseAnalyticsTypes.PurchaseEventParameters;
+import RefundEventParameters = FirebaseAnalyticsTypes.RefundEventParameters;
+import RemoveFromCartEventParameters = FirebaseAnalyticsTypes.RemoveFromCartEventParameters;
+import ScreenViewParameters = FirebaseAnalyticsTypes.ScreenViewParameters;
+import SearchEventParameters = FirebaseAnalyticsTypes.SearchEventParameters;
+import SelectContentEventParameters = FirebaseAnalyticsTypes.SelectContentEventParameters;
+import SelectItemEventParameters = FirebaseAnalyticsTypes.SelectItemEventParameters;
+import SelectPromotionEventParameters = FirebaseAnalyticsTypes.SelectPromotionEventParameters;
+import SetCheckoutOptionEventParameters = FirebaseAnalyticsTypes.SetCheckoutOptionEventParameters;
+import ShareEventParameters = FirebaseAnalyticsTypes.ShareEventParameters;
+import SignUpEventParameters = FirebaseAnalyticsTypes.SignUpEventParameters;
+import SpendVirtualCurrencyEventParameters = FirebaseAnalyticsTypes.SpendVirtualCurrencyEventParameters;
+import UnlockAchievementEventParameters = FirebaseAnalyticsTypes.UnlockAchievementEventParameters;
+import ViewCartEventParameters = FirebaseAnalyticsTypes.ViewCartEventParameters;
+import ViewItemEventParameters = FirebaseAnalyticsTypes.ViewItemEventParameters;
+import ViewItemListEventParameters = FirebaseAnalyticsTypes.ViewItemListEventParameters;
+import ViewPromotionEventParameters = FirebaseAnalyticsTypes.ViewPromotionEventParameters;
+import ViewSearchResultsParameters = FirebaseAnalyticsTypes.ViewSearchResultsParameters;
+import ConsentSettings = FirebaseAnalyticsTypes.ConsentSettings;
+import SettingsOptions = FirebaseAnalyticsTypes.SettingsOptions;
```

Comment:
I'm not sure importing the sub-types into this namespace adds value - it appears that one of them (AnalyticsCallOptions) is used pretty frequently but the rest are used only once for the most part.

So there is a huge block of types up at the top here (which adds to the mental load for the developer + reviewer to remember the local types) without the benefit of really eliminating much extra text (the `FirebaseAnalyticsTypes.` prefix that is eliminated would only show up once anyway...)

I'd prefer fewer new types (created in this namespace by the named imports) and would rather see the types directly referenced as `FirebaseAnalyticsTypes.<subtype>` in the referencing code below